File: primes-trial-division/pascal/nprimes.pas
PROGRAM nPrimes (OUTPUT);

(* CONST NumPrimes = 100; *)
CONST NumPrimes = 1000;
(* CONST NumPrimes = 10000; *)

VAR Count, Index, Found : INTEGER;
    Primes : ARRAY [1..NumPrimes] OF INTEGER;
    
BEGIN

	Found := 0;
	Count := 2;
	
	WHILE Found < NumPrimes DO
	BEGIN

		Found := Found + 1;
		Primes[Found] := Count;
		WRITELN(Count);
		Count := Count + 1;
		
		Index := 1;
		REPEAT
			IF Count MOD Primes[Index] = 0 THEN
			BEGIN
				Count := Count + 1;
				Index := 0;
			END;
			Index := Index + 1;
		UNTIL Index > Found;
	
	END;
	
END.

File: primes-trial-division/fortran/nprimes.f90
       PROGRAM NPRIME
       PARAMETER (N=100)
C      PARAMETER (N=1000)
C      PARAMETER (N=10000)
       INTEGER FOUND, COUNT, I, PRIMES(N)
       
       FOUND = 0
       COUNT = 2   
       
  10   FOUND = FOUND + 1
       WRITE (*,*) COUNT
       PRIMES(FOUND) = COUNT
       COUNT = COUNT + 1
         
       I = 1
       IF (FOUND .LE. N) THEN
  20     IF (MOD(COUNT,PRIMES(I)) .EQ. 0) THEN
           COUNT = COUNT + 1
           I = 1
         ELSEIF (I .EQ. FOUND) THEN
           GOTO 10
         ENDIF
         I = I + 1
         GOTO 20  
       ELSE
         GOTO 30
       ENDIF
         
  30   STOP
       END PROGRAM NPRIME
